Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/Resolving/ResolveTargetTypeLookup.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    [BurstCompatible]
+    internal struct ResolveTargetTypeLookup : IDisposable
+    {
+        private UnsafeParallelHashMap<ResolveTargetID, ResolveTargetWriteData> m_ResolveTargetWriteDataByID;
```

Comment:
I can't unfortunately because I call the methods `Dispose` and `Add`.

In the Class sense, a readonly Dictionary just means that the variable will always be THAT Dictionary. But the contents can change.

In the struct sense like this we can't change anything. Although because it's using pointers behind the scenes it will PROBABLY still work but the code is then littered with warnings like this:

`Possibly impure struct method called on readonly variable: struct value always copied before invocation`

Reply:
Hmm, if a pointer change does occur because of a resize does the stuct remain valid?

For the warnings you could always suppress the warnings for that specific field.

Reply:
Yes, because the struct just wraps the pointer. 

Kind of looks like this structually

```
- Struct
    - Internal Collection Ptr
        - Actual Backing Memory
        - Meta Data  
```

If you copy the struct, the pointer address gets copied along with it so the new copy is still pointing to the original memory.

If a resize happens, only the `Actual Backing Memory` gets replaced and the `Internal Collection Ptr` just points to the new `Actual Backing Memory`.

For the warnings, that would be nice, but I can't seem to find that for a specific field. I can manually do it for every time I access a function on that field which is tedious, or I can blanket set it for the struct or file but then other fields will also not show that warning which could be dangerous. If we can disable for just that specific field since we know the collection struct is safe to manipulate in that way I'd be happy to add it.

Reply:
ah ok, cool. I didn't realize `UnsafeParallelHashMap` kept it's metadata under the pointer. Great!